### PR TITLE
Support ".Ref" targeting packs and .NET Standard 2.1

### PR DESCRIPTION
**Pull request by dsplaisted**
This supports the ".Ref" targeting packs for .NET Core and ASP.NET Core, as well as .NET Standard 2.1.

The core-sdk repo needs to be updated to include these in the `KnownFrameworkReference` items in the bundled versions file.  However, we need to check this logic into the SDK (and flow it into core-sdk) first, because we changed the layout for the .NET Core targeting pack and things would fail if we bundled a version of the targeting pack that the SDK code couldn't understand.  (Note that the layout in this PR isn't necessarily final either.)

There's temporary logic in this PR to update the `KnownFrameworkReference` items to refer to the new targeting packs for testing purposes.  Once we do the round trip through core-sdk, we can remove that logic.

Also note that this PR currently adds a `KnownFrameworkReference` for NETStandard.Library, but that is checked in to core-sdk already (since it's for a new TargetFramework), so once we have a build from core-sdk with the changes from https://github.com/dotnet/core-sdk/pull/477, I will update this PR to remove that workaround.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs [hunk 1]
```diff
@@ -60,38 +60,75 @@ protected override void ExecuteCore()
...
+                            reference.SetMetadata("Visible", "false");
+                            reference.SetMetadata(MetadataKeys.NuGetPackageId, targetingPack.GetMetadata(MetadataKeys.PackageName));
+                            reference.SetMetadata(MetadataKeys.NuGetPackageVersion, targetingPack.GetMetadata(MetadataKeys.PackageVersion));
+
+                            if (!Path.GetFileName(dll).Equals("netstandard.dll", StringComparison.OrdinalIgnoreCase))
+                            {
+                                reference.SetMetadata("Facade", "true");
+                            }
+
+                            referencesToAdd.Add(reference);
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs]: Extrasct some methods here. These large blocks in if and else are hard to follow.

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs [hunk 2]
```diff
@@ -129,10 +133,11 @@ protected override void ExecuteCore()
...
                     !string.IsNullOrEmpty(knownFrameworkReference.RuntimePackNamePatterns))
                 {
+                    var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
                     foreach (var runtimePackNamePattern in knownFrameworkReference.RuntimePackNamePatterns.Split(';'))
                     {
                         string runtimePackRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(
-                            new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath),
+                            runtimeGraph,
                             RuntimeIdentifier,
                             knownFrameworkReference.RuntimePackRuntimeIdentifiers.Split(';'),
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs]: As long as you're adressing feedback from previous PR at the same time, I asked about hoisting this split too.

Comment (nguerrera) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs]: But I think the whole RID matching thing needs some perf work, so feel free to leave this alone.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props [hunk 3]
```diff
@@ -17,6 +17,31 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Import Project="$(NETCoreSdkBundledVersionsProps)" Condition="Exists('$(NETCoreSdkBundledVersionsProps)')" />
 
+  <ItemGroup Condition="'$(UseRefTargetingPacks)' == 'true'">
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: There should be a bug referenced to remove this.

**Comment by nguerrera** [comment]
I think https://github.com/dotnet/sdk/issues/2839 will start to be a real issue with the .Ref targeting pack as there won't be dependencies downloaded.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props (hunk 3 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Filed #2962 for this (not referenced in the code though).
